### DISCUSSION TITLE
Local refinement around a bluff body

### DISCUSSION THREAD

**Discussion by quentinhorn** [xcompact3d/Incompact3d#311] [General]
Hello,

After investigating Xcompact3D capabilities, I found out it is possible to stretch the domain along one direction in order to refine the mesh.
If I understand well the code is using FFT along all dimensions (even on the stretched one), therefore refinement possibilities are obviously restricted.
Since we are investigating turbulences near an immersed body, it would be convenient to be able to refine around the object.

I was wondering if you think it would be reasonable to try implementing a method for more robust refinement purposes?
I was thinking of a domain decomposition, with an higher resolution in the "bluff body subdomain". XCompact3d would then resolve NS in all the subdomain, and guarantee continuity of the solution with boundary conditions and interpolations.

I wanted to know your opinion about it.

Thanks in advance!

**Comment by slaizet** [comment]
Hi,
A domain decomposition is tricky to implement because of the incompressibility condition. Note that we are currently investigating the potential of an iterative Poisson solver which would allow the stretching of the mesh in the three spatial directions. This is still work in progress but this might be an option in the future for more flexibility with the mesh.
Best,
Sylvain
